PR Report Refactor

kghamilton89 opened the pull request:
Explanation in RedMine (see 5297)

rodionos commented:
1. Refactor passive to active: e.g. from 'may be used to notify' to 'can notify' or update or similar.
2. Under 'Launch ATSD Sandbox' section, before the command, organize the action items as a bullet point list. Also, refactor `...should contain` and similar to `Specify the recipient email address in ...`
3. Refactor Link Text in line with   https://developers.google.com/style/link-text, for example, instead of `After initial launch, if Mail Client settings need to be reconfigured, follow these instructions.` use `Reconfigure [Mail Client] if necessary`. Also expand how does the user determine when its necessary to configure mail client?
4. Refactor other link texts throghout the document to use specific titles: `For advanced launch settings refer to this guide.` - this is not line with https://developers.google.com/style/link-text.
5. After the user creates a token in 'Generating OAuth Access Token', where does it go?

Please go through these steps `end-to-end`, exactly as specified in the document. Will you receive the two correct emails for 'axibase' repos?

I also recommend reviewing https://developers.google.com/style, we should be using it more and more. That was a great find at the workshop!

kghamilton89 commented:
With the current configuration, user received **two** sets of reports immediately, and then once each day. I modified the rule so that only **one** set will be delivered after launch. Otherwise, everything works as expected. Article is refactored, waiting to push until we finalize overhaul to avoid conflicts.

kghamilton89 commented:
1. Style refactored to reflect guidelines
2. Action items appropriately organized.
3. Link text reconfigured, Mail Client explanation expanded.
4. Link text reconfigured.
5. OAuth handling explanation explained further.

End-to-end test completed, setting contained in PR here will result in single set of PR reports, and daily PR reports from there.

Style guide incorporated.

rodionos commented on how-to/github/pr-report.md:
````diff
@@ -57,13 +62,13 @@ Watch the sandbox container logs for `All applications started` and `Mail Client
 ```
 
-If **Mail Client** is successfully configured, ATSD will notify:
+If **Mail Client** configuration is successful, ATSD notifies:
````
This is difficult to read. Perhaps, `The log file contains the following message if the Mail Client test is successful:` ... `Otherwise, the log file include a warning:`. Something like that...

kghamilton89 replied:
I agree. Changed to what you've suggested.

rodionos commented on how-to/github/pr-report.md:
````diff
@@ -45,7 +50,7 @@ user=myuser@example.org
 ```
 
-This `mail.properties` file defines the mail server which will send outgoing reports. Replace `server`, `user`, and `password` parameters with legitimate credentials. If your outgoing mail server uses a different port than 587 for SMTP queries, define it as an additional parameter in the `mail.properties` file.
+This file defines the mail server that sends outgoing reports. Replace `server`, `user`, and `password` parameters with legitimate credentials. If the outgoing mail server uses a different port than 587 for SMTP queries, define it as an additional parameter:
````
The file contains mail server connection settings, to be precise.

kghamilton89 replied:
Changed.

rodionos commented on how-to/github/pr-report.md:
```diff
@@ -75,57 +80,57 @@ Upon failure, stop the container:
 ![](./images/test-email.png)
 
-After initial launch, if **Mail Client** settings need to be reconfigured, follow these [instructions](https://axibase.com/docs/atsd/administration/mail-client.html).
+Reconfigure [**Mail Client**](https://axibase.com/docs/atsd/administration/mail-client.html) to modify subscriber list or change mail server. ATSD web interface is accessible at [`https://docker_host:8443/`](https://github.com/axibase/dockers/tree/atsd-sandbox#exposed-ports).
```
Subscriber list is not modified in Mail Client... It's modified in the rule editor > Email Notification tab.

kghamilton89 replied:
This is generally weakly explained as well, several new steps and pictures added.

rodionos commented on how-to/github/pr-report.md:
```diff
@@ -75,57 +80,57 @@ Upon failure, stop the container:
+## Generate OAuth Access Token
 
-## Generating OAuth Access Token
+This tool uses an [OAuth Personal Token](https://blog.github.com/2013-05-16-personal-api-tokens/) to query the API without sending user login information.
```
Which tool is referred to here?

rodionos commented on how-to/github/pr-report.md:
```diff
@@ -75,57 +80,57 @@ Upon failure, stop the container:
 
 Configure the token to grant **read:org** permissions in the **admin:org** section. This scope grants read-only organization access to any user with this token, keep it confidential. For more information about token scopes, see [GitHub Developer Documentation](https://developer.github.com/apps/building-oauth-apps/scopes-for-oauth-apps/).
 
+Define this token in the `TOKEN` parameter of the [launch command](#launch-atsd-sandbox).
```
Specify vs define? I think we're referring to `env` as variables, not as parameters. Should the token be generated in the very first step, if it's needed for the launch command?

kghamilton89 replied:
I agree, I think OAuth section should be in the beginning of the article as it is needed for launch. The current layout is potentially confusing. Also, changed to specify, though I think the difference is minor.

kghamilton89 commented:
Additional edits:
* Removed `SERVER_URL` parameter from launch command and associated instructions. I launched successfully without this parameter.
* Added detailed Subscriber list configuration section, which I think is helpful.
* Changes from comments implemented.

rodionos commented on integrations/github/pr-report.md:
```diff
@@ -2,30 +2,51 @@
-This guide shows how to configure [ATSD](https://axibase.com/docs/atsd/) to produce a daily report with all open Pull Requests across an organization's entire collection of repositories and email it to subscribed users. The report will have two parts: open Pull Requests that are able to be merged and passing all secondary status checks, and open Pull Requests that are not ready to be merged becau ...
+Configure [Axibase Time Series Database](https://axibase.com/docs/atsd/) (ATSD) to produce a daily report with all open Pull Requests across a collection of repositories and email the report to subscribed users. The report has two parts: open Pull Requests passing all status checks, and open Pull Requests failing one or more status checks.
+
+GitHub [webhook services](pr-notification.md) can notify repository owners and administrators of a new Pull Request, but for larger organizations with a large collection of repositories, use the **Daily PR Report** to achieve a wider scope. Configure notifications for a group of subscribers via email with ATSD and the [GitHub v4 API](https://developer.github.com/v4/). Setup takes around 10 minute ...
```
Instead of `for larger organizations with a large collection of repositories` > `for teams with a large collection of repositories`

rodionos replied:
What does it mean - achieve wider scope? Can we be precise here what we want to achieve? Provide a consolidated report on all PR activities?

kghamilton89 replied:
Changed as requested, "wider scope" removed.

rodionos commented on integrations/github/pr-report.md:
```diff
@@ -2,30 +2,51 @@
 ![](./images/pr-report-workflow.png)
 
-## Purpose
+Accomplish complete Pull Request management using programmatic integration leveraging the [GraphQL](https://graphql.org/) API query language, featured in the GitHub API.
```
Remove  sentence 'Accomplish...'. It's marketing before user is convinced.

kghamilton89 replied:
Eased marketing tone.

rodionos commented on integrations/github/pr-report.md:
```diff
@@ -2,30 +2,51 @@
+
+![](./images/read:org.png)
+
+Configure the token to grant **read:org** permissions in the **admin:org** section by checking the appropriate box. This scope grants read-only organization access to any user with this token, so keep it confidential. For more information about token scopes, see [GitHub Developer Documentation](https://developer.github.com/apps/building-oauth-apps/scopes-for-oauth-apps/).
```
> so keep it - rephrase because it sounds too informal

kghamilton89 replied:
Changed.

rodionos commented on integrations/github/pr-report.md:
```diff
@@ -2,30 +2,51 @@
+
+Configure the token to grant **read:org** permissions in the **admin:org** section by checking the appropriate box. This scope grants read-only organization access to any user with this token, so keep it confidential. For more information about token scopes, see [GitHub Developer Documentation](https://developer.github.com/apps/building-oauth-apps/scopes-for-oauth-apps/).
+
+Define the token in the `TOKEN` parameter of the [sandbox launch command](#launch-atsd-sandbox).
```
Copy the token not 'Define'

kghamilton89 replied:
Copy

rodionos commented on integrations/github/pr-report.md:
```diff
@@ -2,30 +2,51 @@
-The `ORGANIZATION` variable should contain the case-sensitive name of the organization whose repositories should be queried when generating the daily report. The `TOKEN` variable should contain the [GitHub OAuth token](#generating-oauth-access-token) which must be generated to query repositories using the GraphQL API. The `SUBSCRIBERS` variable should contain the comma-separated list of email add ...
+Modify the launch command to include legitimate information:
+
+* The `ORGANIZATION` variable specifies the case-sensitive name of the organization to monitor.
```
Organization ... ON GITHUB, not just any org name.

kghamilton89 replied:
Specified that they need to have GitHub page, I also removed the "For more information about creating an organization." and just linked via the word organization to those docs.

kghamilton89 replied:
I got to your next note and saw you wanted it removed completely. So I just removed it at the end.

rodionos commented on integrations/github/pr-report.md:
````diff
@@ -35,17 +56,17 @@ docker run -d -p 8443:8443 \
 ```
 
-For information about creating a new organization, see the [GitHub Help Documentation](https://help.github.com/articles/creating-a-new-organization-from-scratch/).
+> For information about creating an organization, see the [GitHub Help Documentation](https://help.github.com/articles/creating-a-new-organization-from-scratch/).
````
No need to link to info on how to create org on github (users can search google).

kghamilton89 replied:
Removed.

rodionos commented on integrations/github/pr-report.md:
```diff
@@ -75,57 +96,49 @@ Upon failure, stop the container:
+![](./images/alerts-rules.png)
 
-The GitHub API may be accessed [several ways](https://developer.github.com/v3/auth/#other-authentication-methods). This tool uses an [OAuth Personal Token](https://blog.github.com/2013-05-16-personal-api-tokens/) to query the API without sending user login information. Follow these instructions to generate a personal token and store it in the local ATSD instance.
+Search for the `github-daily-pr-status` with the available filters and rule and open the **Rule Editor** by clicking the link in the **Name** column.
```
and rule and open?

kghamilton89 replied:
removed "and rule"

rodionos commented on integrations/github/pr-report.md:
```diff
@@ -75,57 +96,49 @@ Upon failure, stop the container:
-Report delivery will be scheduled for 6:00 PM server local time.
+Report delivery now occurs at 6:00 PM server local time.
 
 ## Notification Payload
```
Rather, Notification Contents

kghamilton89 replied:
Changed.

rodionos commented on integrations/github/pr-report.md:
```diff
@@ -531,24 +548,24 @@ The above JSON result sets will be converted to two outgoing email reports, sent
+## Final Confirmation
 
-Before report delivery, ensure all parameters have been correctly configured:
+Define the following parameters in the launch command and execute to begin reporting:
```
Is final confirmation necessary?

kghamilton89 replied:
Nope

rodionos commented on integrations/github/pr-report.md:
```diff
@@ -531,24 +548,24 @@ The above JSON result sets will be converted to two outgoing email reports, sent
-* GraphQL query targets the appropriate organization. Any organization's public repositories may be queried by GitHub GraphQL API.
+* Location of `mail.properties` file (`volume`).
 
 A sample report from [**Siemens**](https://github.com/siemens) repositories using a wider scope:
```
what is `wider scope`?

kghamilton89 replied:
I removed this example entirely as it shows the old report (both fail and success statuses together in one report). Now only the Apache report remains as it actually shows current functionality.

rodionos commented on integrations/github/pr-report.md:
```diff
@@ -531,24 +548,24 @@ The above JSON result sets will be converted to two outgoing email reports, sent
-Clickable URLs redirect to the Pull Request page.
+Clickable URLs redirect to Pull Request page.
 
 For other GitHub tools developed by Axibase, see the [Use Cases Repository](https://github.com/axibase/atsd-use-cases#github).
```
New use cases link is better, or link to md and it will be replaced automatically.

kghamilton89 replied:
replaced with `README.md` link
